### PR TITLE
Make lookbackDelta a option of QueryEngine

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Comment by roidelapluie** [comment]
@juliusv WDYT?

**Comment by roidelapluie** [comment]
I will think this again. That does not look good.

**Comment by juliusv** [comment]
@roidelapluie It'd be fine with me (modulo the comment about trivial getter methods from the other PR).

@brian-brazil what do you think?

**Comment by roidelapluie** [comment]
@juliusv the getter is there to make sure that other packages can not set the value

**Comment by juliusv** [comment]
Hm ok, I'm not too worried about that, but seems like a valid enough reason.

**Comment by roidelapluie** [comment]
I have used your Zero idea. Setting th query lookback to zero is anyway not helpful in prometheus normal use case. We will log to debug if it is set to 0 and changed to the default.

**Comment by brian-brazil** [comment]
:+1:

File: promql/engine.go [hunk 1]
```diff
@@ -305,6 +306,13 @@ func NewEngine(opts EngineOpts) *Engine {
 		metrics.maxConcurrentQueries.Set(-1)
 	}
 
+	if opts.LookbackDelta == 0 {
+		opts.LookbackDelta = defaultLookbackDelta
+		if l := opts.Logger; l != nil {
+			level.Debug(l).Log("msg", "lookback delta is not zero, setting to default value", "value", defaultLookbackDelta)
```
Comment (juliusv) [inline promql/engine.go]: I think you meant:

```suggestion
			level.Debug(l).Log("msg", "lookback delta is zero, setting to default value", "value", defaultLookbackDelta)
```

**Comment by juliusv** [comment]
:+1: besides last comment

**Comment by roidelapluie** [comment]
Fixed your comment @juliusv

**Comment by juliusv** [comment]
Thanks!

**Pull request merged**